File: app/src/main/java/de/bitb/astroskop/viewbuilder/DialogBuilder.java
package de.bitb.astroskop.viewbuilder;

import android.content.Context;
import android.content.DialogInterface;
import android.support.v7.app.AlertDialog;

import de.bitb.astroskop.R;
import lombok.Getter;
import lombok.Setter;

public class DialogBuilder {

    public static final int NO_BUTTON = 0;

    public void showSimpleDialog(Builder builder) {
        AlertDialog.Builder dialog = new AlertDialog.Builder(builder.getContext(), R.style.dialogStyle)
                .setTitle(builder.getTitleId())
                .setMessage(builder.getMessageId())
                .setPositiveButton(builder.getPositivBtnId(), builder.getPositivListner());
        if (builder.getNegativBtnId() != NO_BUTTON) {
            dialog.setNegativeButton(builder.getNegativBtnId(), builder.getNegativListner());
        }
        if (builder.getNeutralBtnId() != NO_BUTTON) {
            dialog.setNeutralButton(builder.getNeutralBtnId(), builder.getNeutralListner());
        }
        dialog.show();
    }

    public void showDeleteDialog(Context context, DialogInterface.OnClickListener posListner) {
        Builder builder = new Builder(context);
        builder.setTitleId(R.string.dialog_delete_title);
        builder.setMessageId(R.string.dialog_delete_message);
        builder.setPositivBtnId(R.string.delete);
        builder.setNegativBtnId(R.string.cancel);
        builder.setPositivListner(posListner);
        showSimpleDialog(builder);
    }

    @Getter
    @Setter
    private static class Builder {
        private Context context;
        private int titleId;
        private int messageId;
        private int positivBtnId;
        private int negativBtnId;
        private int neutralBtnId;
        private DialogInterface.OnClickListener positivListner;
        private DialogInterface.OnClickListener negativListner;
        private DialogInterface.OnClickListener neutralListner;

        Builder(Context context) {
            this.context = context;
        }
    }

}

File: app/src/main/java/de/bitb/astroskop/viewbuilder/DialogBuilder.kt
package de.bitb.astroskop.viewbuilder

import android.content.Context
import android.content.DialogInterface
import android.support.v7.app.AlertDialog

import de.bitb.astroskop.R
import lombok.Getter
import lombok.Setter

class DialogBuilder {

    fun showSimpleDialog(builder: Builder) {
        val dialog = AlertDialog.Builder(builder.context, R.style.dialogStyle)
                .setTitle(builder.titleId)
                .setMessage(builder.messageId)
                .setPositiveButton(builder.positivBtnId, builder.positivListner)
        if (builder.negativBtnId != NO_BUTTON) {
            dialog.setNegativeButton(builder.negativBtnId, builder.negativListner)
        }
        if (builder.neutralBtnId != NO_BUTTON) {
            dialog.setNeutralButton(builder.neutralBtnId, builder.neutralListner)
        }
        dialog.show()
    }

    fun showDeleteDialog(context: Context, posListner: DialogInterface.OnClickListener) {
        val builder = Builder(context)
        builder.titleId = R.string.dialog_delete_title
        builder.messageId = R.string.dialog_delete_message
        builder.positivBtnId = R.string.delete
        builder.negativBtnId = R.string.cancel
        builder.positivListner = posListner
        showSimpleDialog(builder)
    }

    @Getter
    @Setter
    class Builder internal constructor(val context: Context) {
        var titleId: Int = 0
            set(titleId) {
                field = this.titleId
            }
        var messageId: Int = 0
            set(messageId) {
                field = this.messageId
            }
        var positivBtnId: Int = 0
            set(positivBtnId) {
                field = this.positivBtnId
            }
        var negativBtnId: Int = 0
            set(negativBtnId) {
                field = this.negativBtnId
            }
        var neutralBtnId: Int = 0
            set(neutralBtnId) {
                field = this.neutralBtnId
            }
        var positivListner: DialogInterface.OnClickListener? = null
            set(positivListner) {
                field = this.positivListner
            }
        var negativListner: DialogInterface.OnClickListener? = null
            set(negativListner) {
                field = this.negativListner
            }
        var neutralListner: DialogInterface.OnClickListener? = null
            set(neutralListner) {
                field = this.neutralListner
            }
    }

    companion object {

        val NO_BUTTON = 0
    }

}

